src/main/kotlin/org/rust/ide/folding/RustFoldingBuilder.kt

```diff
@@ -0,0 +1,85 @@
+        return "{...}"
+    }
+
+    override fun buildFoldRegions(root: PsiElement, document: Document, quick: Boolean): Array<out FoldingDescriptor> {
+        if (root !is RustFile) return emptyArray()
+
+        val descriptors: MutableList<FoldingDescriptor> = ArrayList()
+
+        PsiTreeUtil.findChildrenOfType(root, RustBlockExpr::class.java).forEach {
+            val block = it.block
+            if (block != null) {
+                descriptors += FoldingDescriptor(it.node, block.textRange)
+            }
+        }
+        PsiTreeUtil.findChildrenOfType(root, RustImplItem::class.java).forEach {
+            val implBody = it.implBody
+            if (implBody != null) {
+                descriptors += FoldingDescriptor(it.node, implBody.textRange)
+            }
+        }
+        PsiTreeUtil.findChildrenOfType(root, RustStructItem::class.java).forEach {
+            val structDeclArgs = it.structDeclArgs
+            if (structDeclArgs != null) {
+                descriptors += FoldingDescriptor(it.node, structDeclArgs.textRange)
+            }
+        }
+        PsiTreeUtil.findChildrenOfType(root, RustStructExpr::class.java).forEach {
+            descriptors += FoldingDescriptor(it.node, it.structExprBody.textRange)
+        }
+        PsiTreeUtil.findChildrenOfType(root, RustEnumItem::class.java).forEach {
+            val enumBody = it.enumBody
+            descriptors += FoldingDescriptor(it.node, enumBody.textRange)
+        }
+        PsiTreeUtil.findChildrenOfType(root, RustEnumVariant::class.java).forEach {
+            val structDeclArgs = it.enumStructArgs
+            if (structDeclArgs != null) {
+                descriptors += FoldingDescriptor(it.node, structDeclArgs.textRange)
+            }
+        }
+        PsiTreeUtil.findChildrenOfType(root, RustFnItem::class.java).forEach {
+            val fnBody = it.block
+            if (fnBody != null) {
+                descriptors += FoldingDescriptor(it.node, fnBody.textRange)
+            }
+        }
+        PsiTreeUtil.findChildrenOfType(root, RustImplMethodMember::class.java).forEach {
+            val methodBody = it.block
+            if (methodBody != null) {
+                descriptors += FoldingDescriptor(it.node, methodBody.textRange)
+            }
+        }
+        PsiTreeUtil.findChildrenOfType(root, RustTraitMethodMember::class.java).forEach {
+            val methodBody = it.block
+            if (methodBody != null) {
+                descriptors += FoldingDescriptor(it.node, methodBody.textRange)
+            }
+        }
+        PsiTreeUtil.findChildrenOfType(root, RustModItem::class.java).forEach {
+            val rbrace = it.rbrace;
+            if (rbrace != null) {
+                descriptors += FoldingDescriptor(it.node, TextRange(it.lbrace.textOffset, rbrace.textOffset))
+            }
+        }
+        return descriptors.toTypedArray()
+    }
+
+    override fun isCollapsedByDefault(node: ASTNode): Boolean {
```

Comment:
Again, expression body will look a bit cleanly here (there is `alt+Enter` intention to switch between block and expression body).